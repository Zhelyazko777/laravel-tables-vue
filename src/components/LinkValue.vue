<template>
  <a 
    :href="route"
    :title="showTooltipOnHover ? tooltip : ''"
    :data-toggle="showTooltipOnHover ? 'tooltip' : ''"
  >
      <span>{{prefix}}</span>
      <span>{{value}}</span>
      <span>{{suffix}}</span>
  </a>
</template>

<script>
export default {
  props: {
    value: {
      required: true,
      type: String|null,
    },
    route: {
      required: true,
      type: String|null,
    },
    prefix: {
      required: true,
      type: String|null,
    },
    suffix: {
      required: true,
      type: String|null,
    },
    tooltip: {
      required: true,
      type: String|null,
    },
    showTooltipOnHover: {
      required: true,
      type: Boolean|null,
    },
  },
}
</script>