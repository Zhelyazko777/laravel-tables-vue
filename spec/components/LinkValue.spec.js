import { mount } from '@vue/test-utils'
import LinkValue from './../../src/components/LinkValue.vue';

describe('LinkValue.vue Test', () => {
  it('Should render corrrect value and empty suffix and preffix', () => {
    const wrapper = mount(LinkValue, {
      propsData: {
        value: 'test value',
        route: null,
        prefix: null,
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();      
  });

  it('Should render corrrect value and prefix if passed and no suffix', () => {
    const wrapper = mount(LinkValue, {
      propsData: {
        value: 'test value',
        route: null,
        prefix: 'prefix',
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });

    expect(wrapper).toMatchSnapshot();      
  });

  it('Should render corrrect value, suffix and prefix if passed', () => {
    const wrapper = mount(LinkValue, {
      propsData: {
        value: 'test value',
        route: null,
        prefix: 'prefix',
        suffix: 'suffix',
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should render corrrect href', () => {
    const wrapper = mount(LinkValue, {
      propsData: {
        value: null,
        route: 'route',
        prefix: null,
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should render correct tooltip if passed and showTooltipOnHover is true', () => {
    const wrapper = mount(LinkValue, {
      propsData: {
        value: null,
        route: null,
        prefix: null,
        suffix: null,
        tooltip: 'Some tooltip',
        showTooltipOnHover: true,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should not render correct tooltip if passed and showTooltipOnHover is false', () => {
    const wrapper = mount(LinkValue, {
      propsData: {
        value: null,
        route: null,
        prefix: null,
        suffix: null,
        tooltip: 'Some tooltip',
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });
});