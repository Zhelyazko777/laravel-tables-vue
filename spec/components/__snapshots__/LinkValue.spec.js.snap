// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkValue.vue Test Should not render correct tooltip if passed and showTooltipOnHover is false 1`] = `<a title="" data-toggle=""><span></span> <span></span> <span></span></a>`;

exports[`LinkValue.vue Test Should render correct tooltip if passed and showTooltipOnHover is true 1`] = `<a title="Some tooltip" data-toggle="tooltip"><span></span> <span></span> <span></span></a>`;

exports[`LinkValue.vue Test Should render corrrect href 1`] = `<a href="route" title="" data-toggle=""><span></span> <span></span> <span></span></a>`;

exports[`LinkValue.vue Test Should render corrrect value and empty suffix and preffix 1`] = `<a title="" data-toggle=""><span></span> <span>test value</span> <span></span></a>`;

exports[`LinkValue.vue Test Should render corrrect value and prefix if passed and no suffix 1`] = `<a title="" data-toggle=""><span>prefix</span> <span>test value</span> <span></span></a>`;

exports[`LinkValue.vue Test Should render corrrect value, suffix and prefix if passed 1`] = `<a title="" data-toggle=""><span>prefix</span> <span>test value</span> <span>suffix</span></a>`;
